### ISSUE TITLE
Overfitting

### ISSUE THREAD

**Issue by githubiubiu** [junkunyuan/HAP#12]
I replicated HAP on a sufficiently large private reid dataset and indeed achieved significant improvement. However, adamw+lr_decay can cause serious overfitting. Adjusting lr, lr_decay, and drop_path_rate will only cause the map to decrease without causing the overfitting to disappear. Using sgd to replace adamw may not overfit, but the map is lower than adamw. I'm not sure if you have a more effective way to prevent overfitting. I look forward to your help. Thank you very much！

**Comment by junkunyuan** [comment]
> I replicated HAP on a sufficiently large private reid dataset and indeed achieved significant improvement. However, adamw+lr_decay can cause serious overfitting. Adjusting lr, lr_decay, and drop_path_rate will only cause the map to decrease without causing the overfitting to disappear. Using sgd to replace adamw may not overfit, but the map is lower than adamw. I'm not sure if you have a more effective way to prevent overfitting. I look forward to your help. Thank you very much！

Thank you for giving our method a try.

I'm a bit confused about the meaning of "map" in your message. Did you perhaps mean "loss"? 
Additionally, I'm curious to know how you determined that overfitting is occurring.

Could you please provide more details? I would appreciate it.

**Comment by githubiubiu** [comment]
Sorry, in fact, the map is the mAP (mean Average Precision) of the reid model. I observed that overfitting is due to the rapid growth of mAP on the validation set in the first 30 epochs, and then it begins to decline.

**Comment by junkunyuan** [comment]
Alright, I understand.

Have you experimented with reducing the learning rate? For instance, you could try values like 2e-3, 5e-4, or even 1e-4?

Learning rate is the most important hyper-parameters to my experience.
